boxsdk/auth/redis_managed_oauth2.py

```diff
@@ -0,0 +1,76 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+from redis import StrictRedis
+from redis.lock import Lock
+from uuid import uuid4
+from boxsdk import JWTAuth, OAuth2
+
+
+class RedisManagedOAuth2Mixin(OAuth2):
+    """
+    Box SDK OAuth2 subclass.
+    Allows for storing auth tokens in redis.
+
+    :param unique_id:
+        An identifier for this auth object. Auth instances which wish to share tokens must use the same ID.
+    :type unique_id:
+        `unicode`
+    :param redis_server:
+        An instance of a Redis server, configured to talk to Redis.
+    :type redis_server:
+        :class:`Redis`
+    """
+    def __init__(self, unique_id=uuid4(), redis_server=None, *args, **kwargs):
+        self._unique_id = unique_id
+        self._redis_server = redis_server or StrictRedis()
+        super(RedisManagedOAuth2Mixin, self).__init__(*args, **kwargs)
+        if self._access_token is None:
+            self._update_current_tokens()
+        self._refresh_lock = Lock(redis=self._redis_server, name='{0}_lock'.format(self._unique_id))
+
+    def _update_current_tokens(self):
+        self._access_token, self._refresh_token = self._redis_server.hvals(self._unique_id) or (None, None)
+
+    @property
+    def unique_id(self):
+        """
+        Get the unique ID used by this auth instance. Other instances can share tokens with this instance
+        if they share the ID with this instance.
+        """
+        return self._unique_id
+
+    def _get_tokens(self):
+        """
+        Base class override.
+        Gets the latest tokens from redis before returning them.
+        """
+        self._update_current_tokens()
+        return super(RedisManagedOAuth2Mixin, self)._get_tokens()
+
+    def send_token_request(self, *args, **kwargs):
+        """
+        Base class override.
+        Saves the refreshed tokens in redis.
+        """
+        tokens = super(RedisManagedOAuth2Mixin, self).send_token_request(*args, **kwargs)
+        self._redis_server.hmset(self._unique_id, {'access': tokens[0], 'refresh': tokens[1]})
```

Comment:
Whoops. I refactored the base class to make _store_tokens overridable to avoid overriding this, but then forgot to change it.